### PR TITLE
[BACKPORT] Integrations from main

### PR DESCRIPTION
**Pull request by Swiddis**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by pjfitzgibbons — APPROVED** [review]

**Review by mengweieric — APPROVED** [review]

File: common/constants/integrations.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export const OPENSEARCH_DOCUMENTATION_URL = 'https://opensearch.org/docs/latest/integrations/index';
+export const ASSET_FILTER_OPTIONS = ['index-pattern', 'search', 'visualization', 'dashboard'];
+// TODO get this list dynamically from the API
+export const INTEGRATION_TEMPLATE_OPTIONS = ['nginx'];
+// TODO get this list dynamically from the API
+export const INTEGRATION_CATEOGRY_OPTIONS = ['communication', 'http', 'cloud', 'container', 'logs'];
```
Comment (YANG-DB) [inline common/constants/integrations.ts:11]: these fields shouldn’t be hard coded...

File: public/components/integrations/components/__tests__/testing_constants.ts [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    hits: [
+      {
+        name: 'nginx',
+        version: '1.0.1',
+        displayName: 'NginX Dashboard',
+        integrationType: 'logs',
+        description: 'Nginx HTTP server collector',
+        license: 'Apache-2.0',
+        type: 'logs',
+        author: 'John Doe',
```
Comment (YANG-DB) [inline public/components/integrations/components/__tests__/testing_constants.ts:23]: replace to OpenSearch

File: public/components/integrations/components/__tests__/testing_constants.ts [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
```
Comment (YANG-DB) [inline public/components/integrations/components/__tests__/testing_constants.ts:56]: replace to opensearch

File: server/adaptors/integrations/__data__/repository/nginx/nginx-1.0.1.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{
+    "name": "nginx",
+    "version": "1.0.1",
+    "displayName": "NginX Dashboard",
+    "description": "Nginx HTTP server collector",
+    "license": "Apache-2.0",
+    "type": "logs",
+    "author": "OpenSearch",
+    "sourceUrl": "https://github.com/opensearch-project/observability/tree/2.x/integrations/nginx",
```
Comment (YANG-DB) [inline server/adaptors/integrations/__data__/repository/nginx/nginx-1.0.1.json:9]: 2.x -> would have to be changed to main

File: server/adaptors/integrations/__data__/repository/nginx/nginx-1.0.0.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
+{
+    "name": "nginx",
+    "version": "1.0.0",
+    "displayName": "NginX Dashboard",
+    "description": "Nginx HTTP server collector",
+    "license": "Apache-2.0",
+    "type": "logs",
+    "link": "https://www.nginx.com/",
+    "author": "OpenSearch",
+    "sourceUrl": "https://github.com/opensearch-project/observability/tree/2.x/integrations/nginx",
```
Comment (YANG-DB) [inline server/adaptors/integrations/__data__/repository/nginx/nginx-1.0.0.json:10]: 2.x would need to change to main

**Review by YANG-DB — CHANGES_REQUESTED** [review]

File: common/constants/integrations.ts (hunk 1 above)
Comment (Swiddis) [reply common/constants/integrations.ts:11]: Yep, I know, we can fix it later.

File: public/components/integrations/components/__tests__/testing_constants.ts (hunk 2 above)
Comment (Swiddis) [reply public/components/integrations/components/__tests__/testing_constants.ts:23]: This is a test case.

File: public/components/integrations/components/__tests__/testing_constants.ts (hunk 3 above)
Comment (Swiddis) [reply public/components/integrations/components/__tests__/testing_constants.ts:56]: This is a test case.

File: .husky/_/husky.sh [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  sh -e "$0" "$@"
+  exitCode="$?"
+
+  if [ $exitCode != 0 ]; then
+    echo "husky - $hook_name hook exited with code $exitCode (error)"
+    exit $exitCode
+  fi
+
+  exit 0
+fi
```
Comment (ps48) [inline .husky/_/husky.sh:30]: This is not part of main branch

**Review by YANG-DB — APPROVED** [review]

File: common/constants/shared.ts [hunk 7]
```diff
@@ -69,6 +74,7 @@ export const PPL_NEWLINE_REGEX = /[\n\r]+/g;
 
 // Observability plugin URI
 const BASE_OBSERVABILITY_URI = '/_plugins/_observability';
+const BASE_INTEGRATIONS_URI = '/_plugins/_integrations'; // Used later in front-end for routing
```
Comment (ps48) [inline common/constants/shared.ts:77]: I think this is not used as well. We can remove this

**Pull request merged**